### PR TITLE
Add a status flag to the ttemplate table, and allow this status to be…

### PR DESCRIPTION
**Pull request by knoxsp**
… used as a filter for templates which are no longer active

### CODE REVIEW

File: hydra_base/lib/template/__init__.py [hunk 1]
```diff
@@ -538,7 +538,32 @@ def add_child_template(parent_id, name, description=None, **kwargs):
...
+    """
+        Set the status of a template to active
+    """
 
+    _set_template_status(template_id, 'A')
+
+@required_perms("edit_template")
+def deactivate_template(template_id, **kwargs):
+    """
+        Set the status of a template to active
```
Comment (pmslavin) [inline hydra_base/lib/template/__init__.py:563]:
```suggestion
        Set the status of a template to inactive
```

File: hydra_base/lib/template/__init__.py [hunk 2]
```diff
@@ -603,12 +632,14 @@ def delete_template(template_id, **kwargs):
...
 
 @required_perms("get_template")
-def get_templates(load_all=True, **kwargs):
+def get_templates(load_all=True, include_inactive=False, **kwargs):
     """
         Get all templates.
         Args:
             load_all Boolean: Returns just the template entry or the full
             template structure (template types and type attrs)
+            ignore_status Boolean: If true, returns all templates. If false, returns
```
Comment (pmslavin) [inline hydra_base/lib/template/__init__.py:641]: It's called `include_inactive` in the func def and this is the clearer name.

```suggestion
            include_inactive Boolean: If true, returns all templates. If false, returns
```

**Review by pmslavin — COMMENTED** [review]
Looks fine; alembic works for me from `b04cb2e57cb0` ...
```bash
(Hydra_py3.6) [0]paul@paul-VirtualBox:~/code/Hydra_py3.6/hydra-base/hydra_base/db $ alembic upgrade head
24404 2020-12-11 11:42:22,154 - INFO - Registering data type "SCALAR".
24404 2020-12-11 11:42:22,154 - INFO - Registering data type "ARRAY".
24404 2020-12-11 11:42:22,154 - INFO - Registering data type "DESCRIPTOR".
24404 2020-12-11 11:42:22,155 - INFO - Registering data type "DATAFRAME".
24404 2020-12-11 11:42:22,155 - INFO - Registering data type "TIMESERIES".
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade b04cb2e57cb0 -> 1d3ab26415ec, template_status
```
...two micro-changes suggested, both in comments.

File: hydra_base/lib/template/__init__.py (hunk 1 above)
Comment (knoxsp) [reply hydra_base/lib/template/__init__.py]: done

File: hydra_base/lib/template/__init__.py (hunk 2 above)
Comment (knoxsp) [reply hydra_base/lib/template/__init__.py]: done

**Review by pmslavin — APPROVED** [review]

**Pull request merged**